ooniprobe/Model/Database/Result.mm

```diff
@@ -90,8 +83,25 @@ -(NSString*)getLocalizedNetworkType{
     return @"";
 }
 
--(void)addRuntime:(float)value{
-    self.runtime+=value;
+-(Measurement*)getFirstMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] order:@"start_time"] fetch];
+    if ([measurements count] > 0)
+        return [measurements objectAtIndex:0];
+    return nil;
+}
+
+-(Measurement*)getLastMeasurement{
+    SRKResultSet *measurements = [[[[Measurement query] where:@"result_id = ? AND is_rerun = 0" parameters:@[self]] orderByDescending:@"start_time"] fetch];
```

Comment:
Can you set a LIMIT on the query so that you don't retrieve all the rows just to get the first or last one?

Reply:
Great, will do